spec/engine/miq_ae_method_service/miq_ae_service_methods_spec.rb

```diff
@@ -162,5 +162,32 @@ def category_create_script
       ct.reload
       expect(ct.entries.collect(&:name).include?('fred')).to be_truthy
     end
+
+    context "#create_service_provision_request" do
+      let(:options) { {:fred => :flintstone} }
+      let(:svc_options) { {:dialog_style => "medium"} }
+      let(:user) { FactoryGirl.create(:user_with_group) }
+      let(:template) { FactoryGirl.create(:service_template_ansible_playbook) }
+      let(:miq_request) { FactoryGirl.create(:service_template_provision_request) }
+      let(:svc_template) do
+        MiqAeMethodService::MiqAeServiceServiceTemplate.find(template.id)
+      end
+      let(:workspace) do
+        double("MiqAeEngine::MiqAeWorkspaceRuntime",
+               :root               => options,
+               :persist_state_hash => {},
+               :ae_user            => user)
+      end
+      let(:miq_ae_service) { MiqAeService.new(workspace) }
+
+      it "create service request" do
+        allow(workspace).to receive(:disable_rbac)
+        expect(svc_template).to receive(:instance_variable_get).with('@object').and_return(template)
+        expect(template).to receive(:provision_request).with(user, svc_options).and_return(miq_request)
+
+        result = miq_ae_service.execute(:create_service_provision_reques, svc_template, svc_options)
```

Comment:
That is probably why there is no test for ```create_automation_request``` and ```create_provision_request```. Should we move these methods to ```lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service.rb``` ? That would also be consistent with ```$evm.create_notification```. And it would be easier to write tests. The drawback is that it would probably require more changes to ```manageiq-content``` repository.

What do you think @gmcculloug, @mkanoor ?